File: Python/백준/Silver/2870. 수학숙제/수학숙제.py
N = int(input())

lst = []
for _ in range(N):
    S = input()
    num = 0
    flag = False
    for i in range(len(S)):
        if '0' <= S[i] <= '9':
            num = num * 10 + int(S[i])
            flag = True
        elif flag == True:
            lst.append(num)
            num = 0
            flag = False
    if flag:
        lst.append(num)

lst.sort()
for n in lst:
    print(n)

File: Java/백준/Silver/2870. 수학숙제/수학숙제.java
import java.io.*;
import java.util.*;
import java.math.BigInteger;

public class Main {

    public static void main(String[] args) throws Exception {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());

        ArrayList<BigInteger> arrayList = new ArrayList<>();
        for (int i = 0; i < N; i++) {
            String S = br.readLine();
            BigInteger num = new BigInteger("0");
            BigInteger mul = new BigInteger("10");
            boolean flag = false;
            for (int j = 0; j < S.length(); j++) {
                if ('0' <= S.charAt(j) && S.charAt(j) <= '9') {
                    num = mul.multiply(num).add(BigInteger.valueOf(S.charAt(j) - '0'));
                    flag = true;
                } else if (flag) {
                    arrayList.add(num);
                    num = new BigInteger("0");
                    flag = false;
                }
            }

            if (flag) {
                arrayList.add(num);
            }
        }

        Collections.sort(arrayList);

        StringBuilder sb = new StringBuilder();
        for (BigInteger n : arrayList) {
            sb.append(n).append('\n');
        }

        System.out.println(sb);
    }
}